server/schedule/placement/rule_manager.go

```diff
@@ -305,9 +308,39 @@ func (m *RuleManager) GetRulesForApplyRegion(region *core.RegionInfo) []*Rule {
 }
 
 // FitRegion fits a region to the rules it matches.
-func (m *RuleManager) FitRegion(stores StoreSet, region *core.RegionInfo) *RegionFit {
+func (m *RuleManager) FitRegion(storeSet StoreSet, region *core.RegionInfo) *RegionFit {
+	stores := storeSet.GetStores()
```

Comment:
Does it get all stores in the cluster? It looks we only need stores that contain the region's peers.

Reply:
good cactch, updated.

Reply:
I think you need to make `getStoresByRegion` to accept `storeSet` directly to avoid getting all stores here.